zebedee-reader/src/main/java/com/github/onsdigital/zebedee/util/URIUtils.java

```diff
@@ -79,11 +83,129 @@ public static String getLastSegment(String uri) {
         return segments[segments.length - 1];
     }
 
+    public static String removeAfterKeyword(String path, String keyword) {
+        if (isEmpty(path)) {
```

Comment:
You have used `isEmpty` in a lot of these functions, even when it doesn't make sense in some cases. `isEmpty` checks for both nullability and an empty string. In a lot of these new functions, an empty string is still a valid input.

Reply:
yes, it does make sense. I would also argue that checking for null and throwing a new Exception does not add any value for the user here. So if a string is null it should throw the normal exception.